CRM/Event/Form/Participant.php

```diff
@@ -1360,7 +1360,7 @@ public function submit($params) {
     }
 
     // also store lineitem stuff here
-    if ((($this->_lineItem & $this->_action & CRM_Core_Action::ADD) ||
+    if ((($this->_lineItem && $this->_action & CRM_Core_Action::ADD) ||
```

Comment:
This makes me a little nervous as the impact might be to change behaviour - I agree it looks like a bug but sometimes fixing bad code breaks things - have we branched yet?

Reply:
nope

Reply:
I stepped through this in the code & 

1) $this->_lineItem  is always set for paid events & unset for unpaid events and
2) the behaviour is unchanged by this change